ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -2389,6 +2365,19 @@ FEDataManager::buildDiagonalL2MassMatrix(const std::string& system_name)
     return d_fe_data->d_L2_proj_matrix_diag[system_name].get();
 } // buildDiagonalL2MassMatrix
 
+NumericVector<double>*
+FEDataManager::buildGhostedDiagonalL2MassMatrix(const std::string& system_name)
+{
+    if (!d_fe_data->d_L2_proj_matrix_diag_ghost.count(system_name))
+    {
+        std::unique_ptr<NumericVector<double> > M_vec = buildGhostedSolutionVector(system_name)->clone();
+        *M_vec = *buildDiagonalL2MassMatrix(system_name);
+        M_vec->close();
+        d_fe_data->d_L2_proj_matrix_diag[system_name] = std::move(M_vec);
+    }
+    return d_fe_data->d_L2_proj_matrix_diag_ghost[system_name].get();
```

Comment:
Putting this in `d_fe_data` is wrong - this mass matrix depends on the IB partitioning, but `FEData` doesn't know about the IB partitioning. This is going to cause problems with the scratch hierarchy.

Reply:
In retrospect it doesn't make much sense for the other lumped mass matrix to live here either.

Reply:
I will move both out into the `FEDataManager` class.